### PR TITLE
BLEN-305: Refactor project due to suggestments in D16799

### PR DESCRIPTION
**Pull request by bnagirniak**
Refactor c++ part:
- moved project extern_usdhydra to blender/source/render/hydra and renamed to bf_render_hydra
- renamed cmake key: WITH_USDHYDRA -> WITH_HYDRA
- renamed namespace to blender::render::hydra
- moved code from finalEngineGL.cpp/.h to finalEngine.cpp/.h
- moved py api code from engine.cpp to python.cpp

Refactor python part:
- moved python code to release/scripts/modules/hydra.py
- removed storm and addon code, as it became a module.

### CODE REVIEW

File: source/blender/render/hydra/python.cpp [hunk 1]
```diff
@@ -1,89 +1,133 @@
...
+namespace blender::render::hydra {
 
-Engine::Engine(BL::RenderEngine &b_engine, const std::string &delegateId)
-  : b_engine(b_engine)
+static PyObject *init_func(PyObject * /*self*/, PyObject *args)
 {
-  HdRendererPluginRegistry& registry = HdRendererPluginRegistry::GetInstance();
+  LOG(INFO) << "init_func";
+
+  pxr::PlugRegistry::GetInstance().RegisterPlugins(std::string(BKE_appdir_program_dir()) + "/blender.shared/usd");
```
Comment (DagerD) [inline source/blender/render/hydra/python.cpp]: remove `std::` here and further

File: extern/usdhydra/addon/ui.py [hunk 2]
```diff
@@ -1,83 +0,0 @@
-# SPDX-License-Identifier: Apache-2.0
-# Copyright 2011-2022 Blender Foundation
-
-# <pep8 compliant>
-
-import bpy
-
-
-def get_panels():
```
Comment (DagerD) [inline extern/usdhydra/addon/ui.py:9]: seems like we need this function

**Review by DagerD — CHANGES_REQUESTED** [review]
- build_files/cmake/config/blender_full.cmake (86) WITH_USDHYDRA -> WITH_HYDRA
- check for `using namespace pxr/std` -> somewhere it's used, somewhere not, also remove `std::` and `pxr::` if those namespaces used
- check if we shouldn't remove `extern/usdhydra/addon/preferences.AddonPreferences(bpy.types.AddonPreferences)`
- check if we shouldn't remove `extern/usdhydra/addon/logger.py`

File: extern/usdhydra/addon/ui.py (hunk 2 above)
Comment (bnagirniak) [reply extern/usdhydra/addon/ui.py:9]: we need this function in addons

**Comment by bnagirniak** [comment]
> check if we shouldn't remove extern/usdhydra/addon/preferences.AddonPreferences(bpy.types.AddonPreferences)
This has to be moved to addons.
> check if we shouldn't remove extern/usdhydra/addon/logger.py
This can be used in addon, but not in blender module.

**Review by DagerD — APPROVED** [review]

**Pull request merged**